### PR TITLE
Feature/mat 5551

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-5551](https://jira.cms.gov/browse/MAT-5551)
(Optional) Related Tickets:

### Summary

- Reporting tab is only visible for QDM Measures.
- Made updates to the base configuration as well, as the component is accessible via URL. 

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx [hunk 1]
```diff
@@ -212,4 +231,101 @@ describe("PopulationCriteriaHome", () => {
...
+        ]}
+      >
+        <ApiContextProvider value={serviceConfig}>
+          <Route path={["/measures/testMeasureId/edit/base-configuration"]}>
+            <PopulationCriteriaHome />
+          </Route>
+        </ApiContextProvider>
+      </MemoryRouter>
+    );
+    // verifies if the side nav is created and Reportin is not available
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx]: Missing the `g` at the end of reporting. Should be "Base Configuration" in this case.

File: src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx [hunk 2]
```diff
@@ -212,4 +231,101 @@ describe("PopulationCriteriaHome", () => {
...
+    expect(
+      screen.queryByRole("tab", {
+        name: /Base Configuration/i,
+      })
+    ).toBeNull();
+  });
+
+  // tests for QDM measures, since this test case uses a mock which returns QDM measure
+  // all further test cases will have QDMMeasure being returned from Measure store.
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx:280]: What do you think about having a separate test file for QDM?

Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx:280]: Yeah, we can also do that, Honestly, I don't understand why jest test cases are carrying the config from one test case to consecutive test cases, especially when the test suite executes asynchronously.

I don't have a strong opinion either way, but this PopulationCriteriaHome page Is kinda small, so I feel having 2 different test files could be an overkill. 

Now that I think, we can have 2 different test suites 1 for QiCore and the other for QDM, wondering if the mocked modules can be differentiated. I will verify this.

File: src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx (hunk 1 above)
Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx]: Thank you, Will update it.

File: src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx (hunk 2 above)
Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/PopulationCriteriaHome.test.tsx:280]: Scratch my idea of having 2 suites in the same file, that doesn't work for the mocking module at the component level.

File: src/components/editMeasure/populationCriteria/PopulationCriteriaHome.tsx [hunk 3]
```diff
@@ -61,6 +61,18 @@ export function PopulationCriteriaHome() {
     ]);
   }, [groupsBaseUrl, measure?.groups]);
 
+  const renderQdmComponents = () => {
+    if (measure) {
+      if (path.includes("/base-configuration")) {
+        return isQDM() ? <BaseConfiguration /> : history.push("/404");
+      } else if (path.includes("reporting")) {
+        return isQDM() ? <QDMReporting /> : history.push("/404");
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/populationCriteria/PopulationCriteriaHome.tsx:69]: There's been discussions about moving out the model checking to reduce the amount of "if <model>" checks within a component. 

In my head, I had interpreted it to mean having model specific components over model branches within a single component.

Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/PopulationCriteriaHome.tsx:69]: Yes, I understand. but Reporting for QDM and FHIR are at 2 different places. we can reuse the same Reporting component, but in order to display the tab in 2 different places. The branching should happen outside of Component itself.

**Pull request closed without merging**